pkg/cli/node.go

```diff
@@ -467,12 +477,34 @@ func runDecommissionNodeImpl(
 		}
 
 		if !anyActive && replicaCount == 0 {
-			// We now mark the nodes as fully decommissioned.
-			req := &serverpb.DecommissionRequest{
+			// We now drain the nodes in order to close all SQL connections.
+			// Note: iteration is not necessary here since there are no remaining leases
+			// on the decommissioning node after replica transferral.
+			for _, targetNode := range nodeIDs {
+				if targetNode == localNodeID {
+					// Skip the draining step for the node serving the request, if it is a target node.
+					log.Warning(ctx,
+						"node specified with --host should not be a target node; skipped the drain step for this node.",
```

Comment:
It's not necessary that we hit this code path through --host, right? We could use --self and rely on the default --host value? I'd make this warning a bit more generic perhaps (and also avoid the scary "should"): 

"skipping drain step for node %d, decommissioning through the node itself" or something similar.

Reply:
Done.